miner/worker.go

```diff
@@ -786,6 +786,12 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 		if tx == nil {
 			break
 		}
+
+		// OVM Change - set the timestamp on the header to the
```

Comment:
This is the point in the codepath where the transaction is known, although I think this codepath will be ran twice and the `Time` value will end up being set a second time. This will definitely set the timestamp in the evm context, but the timestamp that ends up on the header will end up being the tx that causes the error case `ErrGasLimitReached`. Need to think about this a little more so that the timestamp on the block header matches the timestamp of the tx

Reply:
By checking the length of the transactions, it prevents the problem that i described above.